bin/build-plugin-zip.sh

```diff
@@ -7,23 +7,81 @@ set -e
 cd "$(dirname "$0")"
 cd ..
 
+# Make sure there are no changes in the working tree.  Release builds should be
+# traceable to a particular commit and reliably reproducible.  (This is not
+# totally true at the moment because we download nightly vendor scripts).
+changed=
+if ! git diff --exit-code > /dev/null; then
+	changed="file(s) modified"
+elif ! git diff --cached --exit-code > /dev/null; then
+	changed="file(s) staged"
+fi
+if [ ! -z "$changed" ]; then
+	git status
+	echo "ERROR: Cannot build plugin zip with dirty working tree."
+	echo "       Commit your changes and try again."
+	exit 1
+fi
+
+branch="$(git rev-parse --abbrev-ref HEAD)"
+if [ "$branch" != 'master' ]; then
+	echo "WARNING: You should probably be running this script against the"
+	echo "         'master' branch (current: '$branch')"
+	echo
+	sleep 2
+fi
+
+# Download all vendor scripts
+vendor_scripts=""
+# Using `command | while read...` is more typical, but the inside of the while
+# loop will run under a separate process this way, meaning that it cannot
+# modify $vendor_scripts.  See:  https://stackoverflow.com/a/16855194
+exec 3< <(
+	# minified versions of vendor scripts
+	php bin/get-vendor-scripts.php
+	# non-minified versions of vendor scripts (for SCRIPT_DEBUG)
+	php bin/get-vendor-scripts.php debug
+)
+while IFS='|' read -u 3 url filename; do
+	echo "$url"
+	echo -n " > vendor/$filename ... "
+	curl -s "$url" -o "vendor/_download.tmp.js"
```

Comment:
I think we need to follow redirects here (`-L`).

Finding I have vendor dependencies files written with the content like the following:

```
<p>You are being redirected to <a href="/react-dom@16.0.0-alpha.13/umd/react-dom.development.js">/react-dom@16.0.0-alpha.13/umd/react-dom.development.js</a>
```